MacOS build instructions out of date?

### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

main

### What Operating System are you using?

Mac OS

### If Other OS, please describe

11.6.2 BigSur

### Relevant hardware info

Intel

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Tried to do a Mac build yesterday following the instructions from `docs/build-macos.md` but ran into problems. I'm using MacPorts.

### MacPorts package install command is out-of-date

```
    sudo port -q install ccache cmake fluidsynth libpng libsdl2 \
              libsdl2_net meson opusfile pkg-config python3 \
              libslirp
```

That doesn't work; `pkg-config` needs to be changed to `pkgconfig`, and `python3` to something more concrete, e.g. `python39`.

### Linking fails

I can get the project built, but I'm getting linker errors. I have XCode & the command line tools installed, I have accepted the license agreement, and I can build & link all sorts of stuff with MacPorts and gcc just fine (e.g. glfw, nanovg, Nim projects that embed & compile various C libraries during the build process, etc.)

My gcc version (whatever came with the OS):

```
% gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 13.0.0 (clang-1300.0.29.30)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

And the linker errors I'm getting:

```
% ninja -C build
ninja: Entering directory `build'
[2/6] Linking target dosbox
FAILED: dosbox
c++  -o dosbox dosbox.p/meson-generated_.._version.cpp.o dosbox.p/src_main.cpp.o dosbox.p/src_dosbox.cpp.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,@loader_path/subprojects/libslirp-v4.6.1 -Wl,-rpath,@loader_path/subprojects/fluidsynth-2.2.6 -Wl,-rpath,@loader_path/subprojects/munt-libmt32emu_2_6_1 -Wl,-rpath,/opt/local/lib src/libs/ghc/libghc.a src/libs/loguru/libloguru.a src/libs/PDCurses/libpdcurses.a src/libs/sdlcd/libsdlcd.a src/libs/whereami/libwhereami.a src/cpu/libcpu.a src/debug/libdebug.a src/dos/libdos.a src/libs/decoders/libdecoders.a src/fpu/libfpu.a src/gui/libgui.a src/libs/ppscale/libppscale.a src/hardware/libhardware.a src/libs/nuked/libnuked.a src/libs/residfp/libresidfp.a subprojects/libslirp-v4.6.1/libslirp.0.dylib src/ints/libints.a src/midi/libmidi.a subprojects/fluidsynth-2.2.6/libfluidsynth.dylib subprojects/munt-libmt32emu_2_6_1/libmt32emu.dylib src/misc/libmisc.a src/shell/libshell.a /opt/local/lib/libSDL2.dylib -ldl -framework CoreServices /opt/local/lib/libopusfile.dylib -framework OpenGL /opt/local/lib/libpng16.dylib /opt/local/lib/libSDL2_net.dylib -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreMIDI -framework CoreFoundation -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_deflate", referenced from:
      VideoCodec::FinishCompressFrame() in libhardware.a(hardware.cpp.o)
  "_deflateEnd", referenced from:
      VideoCodec::FinishVideo() in libhardware.a(hardware.cpp.o)
  "_deflateInit2_", referenced from:
      VideoCodec::SetupCompress(int, int) in libhardware.a(hardware.cpp.o)
  "_deflateReset", referenced from:
      VideoCodec::PrepareCompressFrame(int, ZMBV_FORMAT, unsigned char*, unsigned char*, unsigned int) in libhardware.a(hardware.cpp.o)
  "_inflate", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
  "_inflateInit_", referenced from:
      VideoCodec::SetupDecompress(int, int) in libhardware.a(hardware.cpp.o)
  "_inflateReset", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[3/6] Linking target tests/dos_files
FAILED: tests/dos_files
c++  -o tests/dos_files tests/dos_files.p/dos_files_tests.cpp.o tests/dos_files.p/.._subprojects_googletest-release-1.11.0_googletest_src_gtest-all.cc.o tests/dos_files.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock-all.cc.o tests/dos_files.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock_main.cc.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,@loader_path/../subprojects/libslirp-v4.6.1 -Wl,-rpath,@loader_path/../subprojects/fluidsynth-2.2.6 -Wl,-rpath,@loader_path/../subprojects/munt-libmt32emu_2_6_1 -Wl,-rpath,/opt/local/lib src/libs/ghc/libghc.a src/libs/loguru/libloguru.a libdosbox.a src/libs/sdlcd/libsdlcd.a src/cpu/libcpu.a src/debug/libdebug.a src/libs/PDCurses/libpdcurses.a src/dos/libdos.a src/libs/decoders/libdecoders.a src/fpu/libfpu.a src/gui/libgui.a src/libs/ppscale/libppscale.a src/hardware/libhardware.a src/libs/nuked/libnuked.a src/libs/residfp/libresidfp.a subprojects/libslirp-v4.6.1/libslirp.0.dylib src/ints/libints.a src/midi/libmidi.a subprojects/fluidsynth-2.2.6/libfluidsynth.dylib subprojects/munt-libmt32emu_2_6_1/libmt32emu.dylib src/misc/libmisc.a src/libs/whereami/libwhereami.a src/shell/libshell.a -ldl /opt/local/lib/libSDL2.dylib -framework CoreServices /opt/local/lib/libopusfile.dylib -framework OpenGL /opt/local/lib/libpng16.dylib /opt/local/lib/libSDL2_net.dylib -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreMIDI -framework CoreFoundation -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_deflate", referenced from:
      VideoCodec::FinishCompressFrame() in libhardware.a(hardware.cpp.o)
  "_deflateEnd", referenced from:
      VideoCodec::FinishVideo() in libhardware.a(hardware.cpp.o)
  "_deflateInit2_", referenced from:
      VideoCodec::SetupCompress(int, int) in libhardware.a(hardware.cpp.o)
  "_deflateReset", referenced from:
      VideoCodec::PrepareCompressFrame(int, ZMBV_FORMAT, unsigned char*, unsigned char*, unsigned int) in libhardware.a(hardware.cpp.o)
  "_inflate", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
  "_inflateInit_", referenced from:
      VideoCodec::SetupDecompress(int, int) in libhardware.a(hardware.cpp.o)
  "_inflateReset", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[4/6] Linking target tests/drives
FAILED: tests/drives
c++  -o tests/drives tests/drives.p/drives_tests.cpp.o tests/drives.p/.._subprojects_googletest-release-1.11.0_googletest_src_gtest-all.cc.o tests/drives.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock-all.cc.o tests/drives.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock_main.cc.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,@loader_path/../subprojects/libslirp-v4.6.1 -Wl,-rpath,@loader_path/../subprojects/fluidsynth-2.2.6 -Wl,-rpath,@loader_path/../subprojects/munt-libmt32emu_2_6_1 -Wl,-rpath,/opt/local/lib src/libs/ghc/libghc.a src/libs/loguru/libloguru.a libdosbox.a src/libs/sdlcd/libsdlcd.a src/cpu/libcpu.a src/debug/libdebug.a src/libs/PDCurses/libpdcurses.a src/dos/libdos.a src/libs/decoders/libdecoders.a src/fpu/libfpu.a src/gui/libgui.a src/libs/ppscale/libppscale.a src/hardware/libhardware.a src/libs/nuked/libnuked.a src/libs/residfp/libresidfp.a subprojects/libslirp-v4.6.1/libslirp.0.dylib src/ints/libints.a src/midi/libmidi.a subprojects/fluidsynth-2.2.6/libfluidsynth.dylib subprojects/munt-libmt32emu_2_6_1/libmt32emu.dylib src/misc/libmisc.a src/libs/whereami/libwhereami.a src/shell/libshell.a -ldl /opt/local/lib/libSDL2.dylib -framework CoreServices /opt/local/lib/libopusfile.dylib -framework OpenGL /opt/local/lib/libpng16.dylib /opt/local/lib/libSDL2_net.dylib -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreMIDI -framework CoreFoundation -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_deflate", referenced from:
      VideoCodec::FinishCompressFrame() in libhardware.a(hardware.cpp.o)
  "_deflateEnd", referenced from:
      VideoCodec::FinishVideo() in libhardware.a(hardware.cpp.o)
  "_deflateInit2_", referenced from:
      VideoCodec::SetupCompress(int, int) in libhardware.a(hardware.cpp.o)
  "_deflateReset", referenced from:
      VideoCodec::PrepareCompressFrame(int, ZMBV_FORMAT, unsigned char*, unsigned char*, unsigned int) in libhardware.a(hardware.cpp.o)
  "_inflate", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
  "_inflateInit_", referenced from:
      VideoCodec::SetupDecompress(int, int) in libhardware.a(hardware.cpp.o)
  "_inflateReset", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[5/6] Linking target tests/shell_redirection
FAILED: tests/shell_redirection
c++  -o tests/shell_redirection tests/shell_redirection.p/shell_redirection_tests.cpp.o tests/shell_redirection.p/.._subprojects_googletest-release-1.11.0_googletest_src_gtest-all.cc.o tests/shell_redirection.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock-all.cc.o tests/shell_redirection.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock_main.cc.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,@loader_path/../subprojects/libslirp-v4.6.1 -Wl,-rpath,@loader_path/../subprojects/fluidsynth-2.2.6 -Wl,-rpath,@loader_path/../subprojects/munt-libmt32emu_2_6_1 -Wl,-rpath,/opt/local/lib src/libs/ghc/libghc.a src/libs/loguru/libloguru.a libdosbox.a src/libs/sdlcd/libsdlcd.a src/cpu/libcpu.a src/debug/libdebug.a src/libs/PDCurses/libpdcurses.a src/dos/libdos.a src/libs/decoders/libdecoders.a src/fpu/libfpu.a src/gui/libgui.a src/libs/ppscale/libppscale.a src/hardware/libhardware.a src/libs/nuked/libnuked.a src/libs/residfp/libresidfp.a subprojects/libslirp-v4.6.1/libslirp.0.dylib src/ints/libints.a src/midi/libmidi.a subprojects/fluidsynth-2.2.6/libfluidsynth.dylib subprojects/munt-libmt32emu_2_6_1/libmt32emu.dylib src/misc/libmisc.a src/libs/whereami/libwhereami.a src/shell/libshell.a -ldl /opt/local/lib/libSDL2.dylib -framework CoreServices /opt/local/lib/libopusfile.dylib -framework OpenGL /opt/local/lib/libpng16.dylib /opt/local/lib/libSDL2_net.dylib -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreMIDI -framework CoreFoundation -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_deflate", referenced from:
      VideoCodec::FinishCompressFrame() in libhardware.a(hardware.cpp.o)
  "_deflateEnd", referenced from:
      VideoCodec::FinishVideo() in libhardware.a(hardware.cpp.o)
  "_deflateInit2_", referenced from:
      VideoCodec::SetupCompress(int, int) in libhardware.a(hardware.cpp.o)
  "_deflateReset", referenced from:
      VideoCodec::PrepareCompressFrame(int, ZMBV_FORMAT, unsigned char*, unsigned char*, unsigned int) in libhardware.a(hardware.cpp.o)
  "_inflate", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
  "_inflateInit_", referenced from:
      VideoCodec::SetupDecompress(int, int) in libhardware.a(hardware.cpp.o)
  "_inflateReset", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[6/6] Linking target tests/shell_cmds
FAILED: tests/shell_cmds
c++  -o tests/shell_cmds tests/shell_cmds.p/shell_cmds_tests.cpp.o tests/shell_cmds.p/.._subprojects_googletest-release-1.11.0_googletest_src_gtest-all.cc.o tests/shell_cmds.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock-all.cc.o tests/shell_cmds.p/.._subprojects_googletest-release-1.11.0_googlemock_src_gmock_main.cc.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -Wl,-rpath,@loader_path/../subprojects/libslirp-v4.6.1 -Wl,-rpath,@loader_path/../subprojects/fluidsynth-2.2.6 -Wl,-rpath,@loader_path/../subprojects/munt-libmt32emu_2_6_1 -Wl,-rpath,/opt/local/lib src/libs/ghc/libghc.a src/libs/loguru/libloguru.a libdosbox.a src/libs/sdlcd/libsdlcd.a src/cpu/libcpu.a src/debug/libdebug.a src/libs/PDCurses/libpdcurses.a src/dos/libdos.a src/libs/decoders/libdecoders.a src/fpu/libfpu.a src/gui/libgui.a src/libs/ppscale/libppscale.a src/hardware/libhardware.a src/libs/nuked/libnuked.a src/libs/residfp/libresidfp.a subprojects/libslirp-v4.6.1/libslirp.0.dylib src/ints/libints.a src/midi/libmidi.a subprojects/fluidsynth-2.2.6/libfluidsynth.dylib subprojects/munt-libmt32emu_2_6_1/libmt32emu.dylib src/misc/libmisc.a src/libs/whereami/libwhereami.a src/shell/libshell.a -ldl /opt/local/lib/libSDL2.dylib -framework CoreServices /opt/local/lib/libopusfile.dylib -framework OpenGL /opt/local/lib/libpng16.dylib /opt/local/lib/libSDL2_net.dylib -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreMIDI -framework CoreFoundation -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_deflate", referenced from:
      VideoCodec::FinishCompressFrame() in libhardware.a(hardware.cpp.o)
  "_deflateEnd", referenced from:
      VideoCodec::FinishVideo() in libhardware.a(hardware.cpp.o)
  "_deflateInit2_", referenced from:
      VideoCodec::SetupCompress(int, int) in libhardware.a(hardware.cpp.o)
  "_deflateReset", referenced from:
      VideoCodec::PrepareCompressFrame(int, ZMBV_FORMAT, unsigned char*, unsigned char*, unsigned int) in libhardware.a(hardware.cpp.o)
  "_inflate", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
  "_inflateInit_", referenced from:
      VideoCodec::SetupDecompress(int, int) in libhardware.a(hardware.cpp.o)
  "_inflateReset", referenced from:
      VideoCodec::DecompressFrame(unsigned char*, int) in libhardware.a(hardware.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -243,7 +243,7 @@ jobs:
           -Db_lto=true
           -Db_asneeded=true
           -Ddefault_library=static
-          -Dtry_static_libs=opusfile,sdl2,sdl2_net
+          -Dtry_static_libs=opusfile,sdl2,sdl2_net,zlib
           -Dfluidsynth:try-static-deps=true
           build
 

--- a/.github/workflows/windows-meson-msys2.yml
+++ b/.github/workflows/windows-meson-msys2.yml
@@ -10,7 +10,7 @@ env:
   CCACHE_DIR:      "/c/ccache"
   CCACHE_MAXSIZE:  "64M"
   CCACHE_COMPRESS: "true"
-  BUILD_FLAGS: "-Dbuildtype=release -Db_asneeded=true --force-fallback-for=fluidsynth,mt32emu -Dtry_static_libs=fluidsynth,mt32emu,opusfile,png,slirp -Dfluidsynth:try-static-deps=true"
+  BUILD_FLAGS: "-Dbuildtype=release -Db_asneeded=true --force-fallback-for=fluidsynth,mt32emu -Dtry_static_libs=fluidsynth,mt32emu,opusfile,png,slirp,zlib -Dfluidsynth:try-static-deps=true"
   BUILD_RELEASE_DIR: build/release
   BUILD_DEBUGGER_DIR: build/debugger
 

--- a/meson.build
+++ b/meson.build
@@ -238,6 +238,8 @@ opus_dep           = dependency('opusfile',
 threads_dep        = dependency('threads')
 sdl2_dep           = dependency('sdl2', version : '>= 2.0.5',
                                 static : ('sdl2' in static_libs_list))
+zlib_dep           = dependency('zlib', version : '>= 1.2.11',
+                                static : ('zlib' in static_libs_list))
 sdl2_net_dep       = optional_dep
 opengl_dep         = optional_dep
 fluid_dep          = optional_dep

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -42,9 +42,20 @@ option('dynamic_core', type : 'combo',
 # This is NOT guaranteed to work - the end results will vary depending on your
 # OS, installed libraries, and dependencies of those libraries.
 #
-option('try_static_libs', type : 'array',
-       choices : ['fluidsynth', 'mt32emu', 'opusfile', 'png', 'sdl2', 'sdl2_net', 'slirp'], value : [],
-       description : 'Attempt to statically link selected libraries.')
+option('try_static_libs',
+       type : 'array',
+       choices : [
+           'fluidsynth',
+           'mt32emu',
+           'opusfile',
+           'png',
+           'sdl2',
+           'sdl2_net',
+           'slirp',
+           'zlib'],
+       value : [],
+       description : 'Attempt to statically link selected libraries.'
+)
 
 option('unit_tests', type : 'feature', value : 'auto',
        description : 'Build unit tests. Auto skips for release builds.')

--- a/src/hardware/meson.build
+++ b/src/hardware/meson.build
@@ -68,6 +68,7 @@ libhardware = static_library('hardware', libhardware_sources,
                                sdl2_dep,
                                sdl2_net_dep,
                                winsock2_dep,
+                               zlib_dep,
                              ])
 
 libhardware_dep = declare_dependency(link_with : libhardware)

--- a/subprojects/zlib.wrap
+++ b/subprojects/zlib.wrap
@@ -1,0 +1,12 @@
+[wrap-file]
+directory = zlib-1.2.12
+source_url = http://zlib.net/fossils/zlib-1.2.12.tar.gz
+source_filename = zlib-1.2.12.tar.gz
+source_hash = 91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9
+patch_filename = zlib_1.2.12-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/zlib_1.2.12-1/get_patch
+patch_hash = 8ec8344f3fe7b06ad4be768fd416694bc56cb4545ce78b0f1c18b3e72b3ec936
+
+[provide]
+zlib = zlib_dep
+

